### PR TITLE
[18 India] Implements Initial Selection and Draft Round (new game)

### PR DESCRIPTION
**Pull request by EdPovi**
Updates to implement the initial hand selection and following drafting of certs for player hands. 

- [X] Branch is derived from the latest `master`
- [X] Code passes linter with `docker compose exec rack rubocop -a`
- [X] Tests pass cleanly with `docker compose exec rack rake`

### Implementation Notes
- Companies are created to represent share certificates in player hands and IPO Rows
- IPO Rows are currently implemented as a two dimensional array 
- Player Hands are stored with player
- Shares in player hands have buyable set to false, which marks them as "reserved shares" 
- Contents of IPO ROWS and Player Hands are displayed on Timeline
- TODO: Hide player hand information from other players

Stock Round and Operating rounds are not yet implemented.

### CODE REVIEW

File: lib/engine/game/g_18_india/game.rb [hunk 1]
```diff
@@ -13,38 +15,44 @@ class Game < Game::Base
...
         POOL_SHARE_DROP = :none
         SOLD_OUT_INCREASE = false
+        NEXT_SR_PLAYER_ORDER = :first_to_pass
 
         HOME_TOKEN_TIMING = :float
         MUST_BUY_TRAIN = :never
 
         CERT_LIMIT = { 2 => 37, 3 => 23, 4 => 18, 5 => 15 }.freeze
+        CERT_LIMIT_CLOSE1 = { 2 => 33, 3 => 22, 4 => 17, 5 => 13 }.freeze
+        CERT_LIMIT_CLOSE2 = { 2 => 29, 3 => 19, 4 => 15, 5 => 12 }.freeze
```
[the remark this answers is not in the document]
Comment (EdPovi) [reply lib/engine/game/g_18_india/game.rb:47]: I have updated that section and was going to add the following to the next Pull Request:
 ```
       CERT_LIMIT = {
          2 => { 10 => 37, 9 => 33, 8 => 29 },
          3 => { 10 => 23, 9 => 22, 8 => 19 },
          4 => { 10 => 18, 9 => 17, 8 => 15 },
          5 => { 10 => 15, 9 => 13, 8 => 12 },
        }.freeze

        # Modified for corporation share limit
        def cert_limit(entity = nil)
          return @cert_limit if entity.nil?
          return 3 if entity.corporation?

            @cert_limit
        end

        # Modify to only count :private type companies (exclude Warrents and Bonds)
        def num_certs(entity)
          certs = entity.shares.sum do |s|
            s.corporation.counts_for_limit && s.counts_for_limit ? s.cert_size : 0
          end
          certs + entity.companies.count {|c| c.type == :private}
        end
```

File: lib/engine/game/g_18_india/game.rb [hunk 2]
```diff
@@ -61,9 +69,271 @@ class Game < Game::Base
...
+          Engine::Round::Stock.new(self, [
+            Engine::Step::Assign,
+            Engine::Step::BuySellParShares,
+          ])
+        end
+
         def operating_round(round_num)
           Engine::Round::Operating.new(self, [
-            Engine::Step::Bankrupt,
+            # Engine::Step::Bankrupt,  # should not need as there is no bankruptcy in game
```
[the remark this answers is not in the document]
Comment (EdPovi) [reply lib/engine/game/g_18_india/game.rb:336]: I haven't worked on the Operating Round parts of the code much yet, I'm working stepwise into the flow. Currently trying to implement some of the Stock Round Quirks.

File: lib/engine/game/g_18_india/game.rb [hunk 3]
```diff
@@ -61,9 +69,271 @@ class Game < Game::Base
...
+        # When converting a Railroad Bond, there is no refund if share price < 100
+        def railroad_bond_convert_cost
+          if gpir_share_price <= 100
+            0
+          else
+            gpir_share_price - 100
+          end
+        end
+
+        def gpir_share_price
```
[the remark this answers is not in the document]
Comment (EdPovi) [reply lib/engine/game/g_18_india/game.rb:304]: thanks for catching that!

File: lib/engine/game/g_18_india/game.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaeljb) [reply lib/engine/game/g_18_india/game.rb:47]: Sounds good!

**Pull request merged**